sdk/samples/iot/iot_samples_common.h

```diff
@@ -128,9 +128,9 @@ typedef enum
  * @param[in]  name      Enumerated name of the sample.
  * @param[out] env_vars  Pointer to struct that will contain all read-in environment variables. May
  * NOT be NULL.
- * @return     AZ_OK if all environment variables required are successfully read and values stored.
+ * @return     `true` if all environment variables required are successfully read and values stored.
  */
-az_result read_environment_variables(
+bool read_environment_variables(

```

Comment:
Looking at the implementation, this change might be OK, but I will let IoT team chime in and decide.